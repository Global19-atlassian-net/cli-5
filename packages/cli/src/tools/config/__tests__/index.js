import fs from 'fs';

import loadConfig from '../';

import {
  cleanup,
  writeFiles,
  getTempDirectory,
} from '../../../../../../e2e/helpers';

import projects from '../../__fixtures__/projects';

const DIR = getTempDirectory('resolve_config_path_test');

// Removes string from all key/values within an object
const removeString = (config, str) =>
  JSON.parse(
    JSON.stringify(config).replace(new RegExp(str, 'g'), '<<REPLACED>>'),
  );

beforeEach(() => {
  cleanup(DIR);
  jest.resetModules();
});

afterEach(() => cleanup(DIR));

test('should have a valid structure by default', () => {
  writeFiles(DIR, {
    'package.json': `{
      "react-native": {
        "reactNativePath": "."
      }
    }`,
  });
  const config = loadConfig(DIR);
  expect(removeString(config, DIR)).toMatchSnapshot();
});

test('should return dependencies from package.json', () => {
  writeFiles(DIR, {
    'node_modules/react-native-test/package.json': '{}',
    'node_modules/react-native-test/ios/HelloWorld.xcodeproj/project.pbxproj':
      '',
    'package.json': `{
      "dependencies": {
        "react-native-test": "0.0.1"
      },
      "react-native": {
        "reactNativePath": "."
      }
    }`,
  });
  const config = loadConfig(DIR);
  expect(removeString(config, DIR)).toMatchSnapshot();
});
