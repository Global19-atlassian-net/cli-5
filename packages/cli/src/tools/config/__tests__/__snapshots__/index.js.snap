// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have a valid structure by default 1`] = `
Object {
  "commands": Array [],
  "dependencies": Object {},
  "haste": Object {
    "platforms": Array [],
    "providesModuleNodeModules": Array [],
  },
  "platforms": Object {
    "android": Object {},
    "ios": Object {},
  },
  "reactNativePath": ".",
  "root": "<<REPLACED>>",
}
`;

exports[`should return dependencies from package.json 1`] = `
Object {
  "commands": Array [],
  "dependencies": Object {
    "react-native-test": Object {
      "assets": Array [],
      "hooks": Object {},
      "params": Array [],
      "platforms": Object {
        "android": null,
        "ios": Object {
          "folder": "<<REPLACED>>/node_modules/react-native-test",
          "libraryFolder": "Libraries",
          "pbxprojPath": "<<REPLACED>>/node_modules/react-native-test/ios/HelloWorld.xcodeproj/project.pbxproj",
          "plist": Array [],
          "podfile": null,
          "podspec": null,
          "projectName": "HelloWorld.xcodeproj",
          "projectPath": "<<REPLACED>>/node_modules/react-native-test/ios/HelloWorld.xcodeproj",
          "sharedLibraries": Array [],
          "sourceDir": "<<REPLACED>>/node_modules/react-native-test/ios",
        },
      },
    },
  },
  "haste": Object {
    "platforms": Array [],
    "providesModuleNodeModules": Array [],
  },
  "platforms": Object {
    "android": Object {},
    "ios": Object {},
  },
  "reactNativePath": ".",
  "root": "<<REPLACED>>",
}
`;
